src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMapping.java

```diff
@@ -290,15 +289,16 @@ private void validateVirtualPhysicalCompatibility() {
     void validateSecondaryIndexes(DynamoTableDescription virtualTable,
                                   DynamoTableDescription physicalTable,
                                   DynamoSecondaryIndexMapper secondaryIndexMapper) {
-        virtualTable.getSis().stream().map(virtualSi -> {
+        checkArgument(virtualTable.getSis().size() == virtualTable.getSis().stream().map(virtualSi -> {
             try {
                 return secondaryIndexMapper.lookupPhysicalSecondaryIndex(virtualSi, physicalTable);
             } catch (MappingException e) {
                 throw new IllegalArgumentException("failure mapping virtual to physical " + virtualSi.getType()
                     + ": " + e.getMessage() + ", virtualSiPrimaryKey=" + virtualSi + ", virtualTable="
                     + virtualTable + ", physicalTable=" + physicalTable);
             }
-        }).collect(Collectors.toMap(DynamoSecondaryIndex::getIndexName, identity()));
+        }).collect(Collectors.toSet()).size(),
+            "More than one virtual secondary index maps to a single physical secondary index");
```

Comment:
"a single" -> "the same"?

Reply:
fixed in c2f3186676fab84f063be727674550c61347cff4